dll/win32/shell32/wine/classes.c

```diff
@@ -44,7 +44,35 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(shell);
 
-#define MAX_EXTENSION_LENGTH 20
+#define MAX_EXTENSION_LENGTH 20 // FIXME: The limit is 254?
+
+HRESULT HCR_GetProgIdKeyOfExtension(PCWSTR szExtension, PHKEY phKey, BOOL AllowFallback)
+{
+    LONG err, cb;
+    WCHAR ext[max(1 + MAX_EXTENSION_LENGTH + 1, MAX_PATH)];
+    WCHAR progid[MAX_PATH];
+    if (szExtension[0] != '.')
+    {
+        ext[0] = '.';
+        lstrcpynW(ext + 1, szExtension, _countof(ext) - 1);
+        szExtension = ext;
+    }
+    cb = sizeof(progid);
+    err = RegQueryValueW(HKEY_CLASSES_ROOT, szExtension, progid, &cb);
```

Comment:
Not using the Ex function with validation on the data type? (the default value is usually REG_SZ, but not always.)

Reply:
Yes, you are correct, we should be checking the type.

Less than half the code in shell32 is probably checking the type correctly and certainly no code in this file checks anything at all. But to be pedantic, using the Ex function is not enough because the string is not guaranteed to be null terminated. `RegGetValueW` on the other hand is documented to properly null terminate strings. Our implementation however does not but I will file a bug.